.github/workflows/build-test-osx-arm64.yaml

```diff
@@ -45,16 +45,18 @@ jobs:
       - uses: actions/checkout@v3
         with:
           submodules: true
-      - name: Restore Opam Cache
-        id: restore-opam-cache
-        uses: actions/cache/restore@v3
+      # Note that this actions does cache read and cache write.
+      - name: Cache Opam
+        uses: actions/cache@v3
         if: ${{ inputs.use-cache }}
         env:
           SEGMENT_DOWNLOAD_TIMEOUT_MINS: 2
         with:
           path: ~/.opam
           #TODO: we should add the md5sum of opam.lock as part of the key
-          key: "${{ runner.os }}-${{ runner.arch }}-${{ env.OPAM_SWITCH_NAME }}-opam-deps"
+          key: ${{ runner.os }}-${{ runner.arch }}-${{ env.OPAM_SWITCH_NAME }}-opam-deps-${{ github.run_id }}
```

Comment:
Again, please add comments in the code rather than the commit message.
I have no idea how this key: restore-keys work, and it does not seem intuitive to me at all, so a small
explanation would be useful here.

Reply:
Also you went back and forth on this using actions/cache vs actions/cache/restore but in the process you removed one of my comment that was explaining how actions/cache works (and the fast that there was a post-cache phase at the end that was saving things).
Please restore my old comment, so in 6 months someone from PA can understand what's going on.